bundles/org.openhab.core/src/main/java/org/openhab/core/i18n/I18nException.java

```diff
@@ -0,0 +1,171 @@
+        setupI18n(bundle, i18nProvider, Locale.ENGLISH);
+    }
+
+    /**
+     * Setup the internationalization support
+     *
+     * In case the message starts with "@text/", one of both setupI18n() methods needs to be called before using the
+     * getMessage() and getLocalizedMessage() methods.
+     *
+     * The getLocalizedMessage() method will then consider the language provided by the locale parameter.
+     *
+     * @param bundle the bundle containing the i18n properties
+     * @param i18nProvider the translation provider
+     * @param locale the language to use for localizing the message
+     */
+    public void setupI18n(Bundle bundle, TranslationProvider i18nProvider, Locale locale) {
+        this.bundle = bundle;
+        this.i18nProvider = i18nProvider;
+        this.locale = locale;
+    }
+
+    @Override
+    public @Nullable String getMessage() {
+        Bundle localeBundle = this.bundle;
+        TranslationProvider localI18nProvider = this.i18nProvider;
+        if (msgKey.isBlank()) {
+            return super.getMessage();
+        } else if (localeBundle == null || localI18nProvider == null) {
+            logger.warn("Internationalization support is not setup for the exception");
+            return super.getMessage();
+        } else {
+            return localI18nProvider.getText(localeBundle, msgKey, null, Locale.ENGLISH, msgParams);
+        }
+    }
+
+    @Override
+    public @Nullable String getLocalizedMessage() {
+        Bundle localeBundle = this.bundle;
+        TranslationProvider localI18nProvider = this.i18nProvider;
+        Locale localLocale = this.locale;
+        if (msgKey.isBlank()) {
+            return super.getMessage();
+        } else if (localeBundle == null || localI18nProvider == null) {
+            logger.warn("Internationalization support is not setup for the exception");
+            return super.getMessage();
+        } else {
+            return localI18nProvider.getText(localeBundle, msgKey, null,
+                    localLocale != null ? localLocale : Locale.ENGLISH, msgParams);
+        }
+    }
+
+    /**
+     * Provides the untranslated message
+     *
+     * If the message does not start with "@text/", it returns the same as the getMessage() method.
+     * If the message starts with "@text/" and no optional arguments are set, it returns a string of this
+     * kind: @text/key
+     * If the message starts with "@text/" and optional arguments are set, it returns a string of this kind: @text/key [
+     * "param1", "param2" ]
+     *
+     * @return the untranslated message or null if the message is undefined
+     */
+    public @Nullable String getUntranslatedMessage() {
+        if (msgKey.isBlank()) {
+            return super.getMessage();
+        }
+        String result = "@text/" + msgKey;
+        Object @Nullable [] params = msgParams;
+        if (params != null && params.length > 0) {
```

Comment:
You can reduce this section to a one liner by using the `Collectors.joining(CharSequence delimiter, CharSequence prefix, CharSequence suffix)` method.

Reply:
Done